contracts/service-registry/src/state.rs

```diff
@@ -143,12 +143,46 @@ pub enum AuthorizationState {
     Jailed,
 }
 
-type ServiceName = str;
+pub struct VerifierPerChainIndexes<'a> {
+    pub verifier_address: MultiIndex<
+        'a,
+        (ServiceName, VerifierAddress),
+        (),
+        (ServiceName, ChainName, VerifierAddress),
+    >,
+}
+
+impl<'a> IndexList<()> for VerifierPerChainIndexes<'a> {
+    fn get_indexes(&'_ self) -> Box<dyn Iterator<Item = &'_ dyn Index<()>> + '_> {
+        let v: Vec<&dyn Index<()>> = vec![&self.verifier_address];
+        Box::new(v.into_iter())
+    }
+}
+
+pub const VERIFIERS_PER_CHAIN: IndexedMap<
+    (ServiceName, ChainName, VerifierAddress),
```

Comment:
is it not possible to use references for keys anymore?

Reply:
Yeah it is! Changed to references.